draft-ietf-jose-json-web-proof.md

```diff
@@ -451,6 +451,29 @@ This example flattened JSON serialization shows the presentation form with both
 <{{./fixtures/build/bbs-holder.json.jwp.wrapped}}>
 Figure: JSON Serialization of Presentation
 
+# Encrypted JSON Web Proofs
+
+Access to JWPs containing non-public material
+by parties without legitimate access to the non-public information
+MUST be prevented.
+This can be accomplished by encrypting the JWP
+when potentially observable by such parties
+to prevent the disclosure of private information.
+The use of an Encrypted JWP, which is a JWE [@!RFC7516]
+with a JWP as its plaintext value,
+is recommended for this purpose.
+The processing of Encrypted JWPs is identical to
+the processing of other JWEs.
+
+A `cty` (content type) Header Parameter value of `jwp` is used to indicate
+that the content of the JWE is a JWP using the JWP Compact Serialization.
+A `cty` (content type) Header Parameter value of `jwp+json` is used to indicate
+that the content of the JWE is the UTF-8 encoding of a JWP using the JWP JSON Serialization.
+In both cases, the `cty` (content type) Header Parameter MUST be present
+unless the application knows that the encrypted content is
+a JWP by another means or convention,
+in which case the `cty` value would typically be omitted.
```

Comment:
These seem like they should be identical to the guidance in JWP around `typ` - if present, they SHOULD be the effective media type of the contained JWP.  This MAY be in the JWP as the `typ` protected header parameter, or omitted from the JWP based on context.

Reply:
We would need to define two new "typ" values saying that the whole thing is an encrypted JWT using one or the other serializations if we were to go that route.  But it would be better to use application-defined "typ" values using the +jwp suffix and/or application-defined "typ" values using the JSON Serialization.

Instead, the encryption treatment is intentionally exactly the same as for JWKs.  Compare the language in this PR with that at https://www.rfc-editor.org/rfc/rfc7517.html#section-7 and let me know if there are changes you'd like to make it even more paralle.

Reply:
Oh wait - I get it now.  Now that we have explicit typing, we should use the "typ" value of the enclosed JWP as the JWE "cty" value.  I'll take a stab at that.

Reply:
@dwaite I updated the description to cater to explicit typing.  Please re-review.